pkg/controller/catalog/storage/mem_storage.go

```diff
@@ -16,211 +16,45 @@ limitations under the License.
 
 package storage
 
-import (
-	"fmt"
-	"log"
-	"strings"
-
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
-)
-
-type bindingPair struct {
-	binding    *model.ServiceBinding
-	credential *model.Credential
-}
-
 type memStorage struct {
-	brokers map[string]*model.ServiceBroker
-	// This gets fetched when a SB is created (or possibly later when refetched).
-	// It's static for now to keep compatibility, seems like this could be more dynamic.
-	catalogs map[string]*model.Catalog
-	// maps instance ID to instance
-	services map[string]*model.ServiceInstance
-	// maps binding ID to binding
-	// TODO: support looking up all bindings for a service instance.
-	bindings map[string]*bindingPair
+	brokers        *memStorageBroker
+	serviceClasses *memStorageServiceClass
+	instances      map[string]*memStorageInstance
+	bindings       map[string]*memStorageBinding
 }
 
 // CreateMemStorage creates an instance of Storage interface, backed by memory.
 func CreateMemStorage() Storage {
 	return &memStorage{
-		brokers:  make(map[string]*model.ServiceBroker),
-		catalogs: make(map[string]*model.Catalog),
-		services: make(map[string]*model.ServiceInstance),
-		bindings: make(map[string]*bindingPair),
+		brokers:        newMemStorageBroker(),
+		serviceClasses: newMemStorageServiceClass(),
+		instances:      make(map[string]*memStorageInstance),
+		bindings:       make(map[string]*memStorageBinding),
 	}
 }
 
-func (s *memStorage) GetInventory() (*model.Catalog, error) {
-	services := []*model.Service{}
-	for _, v := range s.catalogs {
-		services = append(services, v.Services...)
-	}
-	return &model.Catalog{Services: services}, nil
-}
-
-func (s *memStorage) ListBrokers() ([]*model.ServiceBroker, error) {
-	b := []*model.ServiceBroker{}
-	for _, v := range s.brokers {
-		b = append(b, v)
-	}
-	return b, nil
+func (m *memStorage) Brokers() BrokerStorage {
```

Comment:
Yes... it looks odd. @arschles worked on this particular piece, so I won't speak for him, but I can make a good guess that this is k8s-inspired. In k8s, for instance, you get a pod client like so: `kubeClient.Pods(namespace)`. While one might look at that and intuitively think it returns pods from the specified namespace, it actually returns a _client_ for dealing with pods _in_ that namespace. `Brokers()` returning `BrokerStorage` instead of `[]servicecatalog.Broker` feels like the same pattern.

This being said, I'm only trying to rationalize the _why_. There's still room to discuss whether this should change and I think @arschles should weigh in.